### PR TITLE
Added ability to add and edit students, provided validation.

### PR DESCRIPTION
**Pull request by kmusienko**

### CODE REVIEW

File: src/main/java/com/softserve/teamproject/service/impl/StudentServiceImpl.java [hunk 1]
```diff
@@ -52,4 +85,60 @@ public void setStudentResourceAssembler(
...
+   */
+  @Override
+  public Iterable<StudentResource> addStudents(List<Student> students, Integer groupId,
+      String userName) {
+    Group group = groupRepository.findOne(groupId);
+    studentValidator.checkCoordinatorLocationToManipulateStudent(group, userName);
+    for (Student student : students) {
+      student.setGroup(group);
+      if (student.getEnglishLevel() == null) {
+        student.setEnglishLevel(englishLevelRepository.findOne(3));
```
Comment (thedikiy) [inline src/main/java/com/softserve/teamproject/service/impl/StudentServiceImpl.java]: magic number

File: src/main/java/com/softserve/teamproject/validation/impl/StudentValidatorImpl.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    this.studentRepository = studentRepository;
+  }
+
+  @Autowired
+  public void setUserRepository(UserRepository userRepository) {
+    this.userRepository = userRepository;
+  }
+
+  @Override
+  public void checkStudentFields(List<Student> students) {
```
Comment (thedikiy) [inline src/main/java/com/softserve/teamproject/validation/impl/StudentValidatorImpl.java]: i fill like method name must be like fillNotUpdatedFields or smth like that. No validation logic here

File: src/main/java/com/softserve/teamproject/validation/impl/StudentValidatorImpl.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      for (Field field : studentClass.getDeclaredFields()) {
+        field.setAccessible(true);
+        try {
+          if (field.get(student) == null) {
+            Field existedField = existedStudent.getClass().getDeclaredField(field.getName());
+            existedField.setAccessible(true);
+            field.set(student, existedField.get(existedStudent));
+          }
+        } catch (IllegalAccessException | NoSuchFieldException e) {
+          e.printStackTrace();
```
Comment (thedikiy) [inline src/main/java/com/softserve/teamproject/validation/impl/StudentValidatorImpl.java]: Not good to ignore exception, we may use unfilled object

File: src/main/java/com/softserve/teamproject/controller/StudentController.java [hunk 4]
```diff
@@ -41,4 +54,21 @@ public void setStudentService(StudentService studentService) {
...
+  public Iterable<StudentResource> addStudents(@RequestBody List<Student> students,
+      @PathVariable Integer id, Principal principal) {
+    return studentService.addStudents(students, id, principal.getName());
+  }
+
+  @PutMapping(value = "/students", produces = "application/json")
+  @ApiOperation(value = "Edit students for given group", response = Student.class,
+      responseContainer = "List")
+  public Iterable<StudentResource> editStudents(@RequestBody List<Student> students,
+      Principal principal) {
```
Comment (Crutovel) [inline src/main/java/com/softserve/teamproject/controller/StudentController.java]: We can use findLoggedInUsername() from SecurityService instead Principal.

**Pull request merged**